ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -230,15 +230,32 @@ func (scbuilder *SiteConfigBuilder) instantiateCR(target string, originalTemplat
 
 	// Sanity-check the resulting metadata to ensure it's valid compared to the non-overridden original CR:
 	originalMetadata := originalCR["metadata"].(map[string]interface{})
+
+	// Sanity-check the overridden metadata object to ensure that it's instantiated correctly
+	if overriddenCR["metadata"] == nil {
+		return override, fmt.Errorf("Overriden template metadata in %q is not specified!", overridePath)
+	}
 	overriddenMetadata := overriddenCR["metadata"].(map[string]interface{})
+
 	for _, field := range []string{"name", "namespace"} {
 		if originalMetadata[field] != overriddenMetadata[field] {
 			return overriddenCR, fmt.Errorf("Overridden template metadata.%s %q does not match expected value %q", field, overriddenMetadata[field], originalMetadata[field])
 		}
 	}
 	originalAnnotations := originalMetadata["annotations"].(map[string]interface{})
+
+	// Sanity-check the overridden metadata annotations
+	if overriddenMetadata["annotations"] == nil {
+		return override, fmt.Errorf("Overriden template metadata annotations in %q is not specified!", overridePath)
+	}
 	overriddenAnnotations := overriddenMetadata["annotations"].(map[string]interface{})
+
+	// Sanity-check the argocd annotation
 	argocdAnnotation := "argocd.argoproj.io/sync-wave"
+	if overriddenAnnotations[argocdAnnotation] == nil {
+		return override, fmt.Errorf("Overriden template metadata argocd annotation in %q is not specified!", overridePath)
+	}
```

Comment:
Should this be allowed if the `originalAnnotations[argocdAnnotation]` is also nil?